src/test/java/org/broadinstitute/hellbender/tools/picard/sam/SortSamIntegrationTest.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.hellbender.tools.picard.sam;
+
+import htsjdk.samtools.ValidationStringency;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.SamAssertionUtils;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class SortSamIntegrationTest extends CommandLineProgramTest {
+
+    @DataProvider(name="sortbams")
+    public Object[][] sortBAMData() {
+        return new Object[][] {
+                {"count_reads.bam", "count_reads_sorted.bam", null, ".bam", "coordinate"},
+                {"count_reads.bam", "count_reads_sorted.bam", "count_reads.fasta", ".cram", "coordinate"},
+                {"count_reads.bam", "count_reads.bam", null, ".bam", "queryname"},
+                // next two require fix to https://github.com/samtools/htsjdk/issues/365

```

Comment:
here are 2 more disabled tests that need to be mentioned in that ticket (see above)